File: test_cases/python/claude/09.py
def run_tests(solution_module):
    f = solution_module.isNumber
    # positive decimal
    assert f('+3.14') is True
    # exponent with sign
    assert f('3e+7') is True
    # double sign is invalid
    assert f('--6') is False
    # just a dot is invalid
    assert f('.') is False
    # integer string
    assert f('123') is True

File: test_cases/c/claude/09.c
#include <assert.h>
#include <stdbool.h>
#include <stddef.h>

bool isNumber(char* s);

int main(void) {
    assert(isNumber("0") == true);
    assert(isNumber("e") == false);
    assert(isNumber("-123.456e789") == true);
    assert(isNumber(".") == false);
    assert(isNumber("+6e-1") == true);
    return 0;
}